src/net_processing.cpp

```diff
@@ -1277,14 +1280,14 @@ void PeerManagerImpl::MaybeSetPeerAsAnnouncingHeaderAndIDs(NodeId nodeid)
             // As per BIP152, we only get 3 of our peers to announce
             // blocks using compact encodings.
             m_connman.ForNode(lNodesAnnouncingHeaderAndIDs.front(), [this](CNode* pnodeStop){
-                m_connman.PushMessage(pnodeStop, CNetMsgMaker(pnodeStop->GetCommonVersion()).Make(NetMsgType::SENDCMPCT, /*high_bandwidth=*/false, /*version=*/CMPCTBLOCKS_VERSION));
+                PushMessage(*pnodeStop, NetMsg::Make(NetMsgType::SENDCMPCT, /*high_bandwidth=*/false, /*version=*/CMPCTBLOCKS_VERSION));
```

Comment:
Calling `NetMsg::Make` within `PushMessage` seems unnecessary; just add an extra `PeerManagerImpl` helper:

```c++
    template <typename... Args>
    void PushMessage(CNode& node, std::string msg_type, Args&&... args) const
    {
        PushMessage(node, NetMsg::Make(std::move(msg_type), args...));
    }
```

cf https://github.com/ajtowns/bitcoin/commit/0fe129b1b437d39221cef843f969f1adc31b543e

Reply:
Could give it a distinct name like `void MakeAndPushMessage(node, msg_type, args)` to make that clear, I guess?